dev-docs/RFCs/v8.0/binary-support-improvements-rfc.md

```diff
@@ -0,0 +1,186 @@
+
+Before v7, once the data is downloaded, the application is still required to construct a JavaScript array from the binary, adding memor & CPU time tolls on the client side. In v7.x ([Binary Data RFC](v7.x-binary/binary-data-rfc.md)), we added two ways to use binary data that completely circumvent this step, hence eliminating the performance hit in step 1.
+
+* index-based accessor:
+
+    ```js
+    // binaryData is packed on the server, in the format of [x, y, r, g, b, a]
+    const DATA = {src: binaryData, length: binaryData.length / 6};
+
+    new ScatterplotLayer({
+      data: DATA,
+      getPosition: (object, {index, data, target}) => {
+        target[0] = data.src[index * 6];
+        target[1] = data.src[index * 6 + 1];
+        target[2] = 0;
+        return target;
+      },
+      getColor: (object, {index, data, target}) => {
+        target[0] = data.src[index * 6 + 2] * 255;
+        target[1] = data.src[index * 6 + 3] * 255;
+        target[2] = data.src[index * 6 + 4] * 255;
+        target[3] = data.src[index * 6 + 5] * 255;
+        return target;
+      },
+      ...
+    })
+    ```
+
+    This method involves writing an accessor that reads relevant information out of the binary blob at each object index. During attribute update, an internal buffer is allocated, and the accessor is called `numInstance` times to populate it. Hence the performance is comparable to using plain JavaScript array data at step 2 and 3.
+
+* supplying external buffers directly:
+
+    ```js
+    // binaryData is packed on the server, in the format of [x, y, r, g, b, a]
+    const DATA = {
+      length: binaryData.length / 6,
+      attributes: {
+        instancePositions: {value: binaryData, size: 2, stride: 4 * 6, offset: 0},
+        instanceColors: {value: binaryData, size: 4, stride: 4 * 6, offset: 4 * 2, normalized: true}
+      }
+    };
+
+    new PointCloudLayer({
+      data: DATA,
+      getNormal: [0, 0, 0]
+    });
+    ```
+
+    This method skips step 2 entirely, and is possibly the most performant we'll ever get. However, it requires the application (BE and FE) to have knowledge of the internal implementation of a layer, including attribute names, array types and layouts, which is not documented and prone to breakage between minor releases.
+
+
+## Goals
+
+Moving into v8.x, we want to make binary data a first-class citizen, in the following ways:
+
+* All core layers accept a variety of binary data inputs, without exposing the internal layer implementation
+* Binary data are directly uploaded to the GPU if possible, and basic packing operations (e.g. position interleving, transform matrix construction) are performed on the GPU instead of CPU
+
+
+## Proposal
+
+Adding an additional override to the `accessor` prop type that is a string. `props.data[accessor]` should yield a loaders.gl-compatible "attribute descriptor" object.
```

Comment:
String-valued accessors: This is an unused overload for accessors that is potentially a desirable integration point for any one of multiple features, is it worth "spending it" on this feature? As an example, I have a bunch of RFCs that instead propose interpreting string-value accessors as GLSL snippets. Those RFCs should of course be re-evaluated against the GPUTable proposal but I would not consider them completely ice-boxed, and I don't think it is too hard to imagine other interpretations of string valued accessors.

Perhaps first think through if the proposed `accessor` to `data[accessor]` indirection necessary? 
- What does it gain compared to overloading accessors to accept object values that are accessors?
- Considered just supporting `props.data.accessors[<layer accessor name>]` without any need to supply indirection strings via the existing accessors?
- ...

Reply:
Updated based on discussion